src/jdk.compiler/share/classes/com/sun/tools/javac/main/JavaCompiler.java

```diff
@@ -1819,11 +1819,7 @@ public void close() {
             for (Closeable c: closeables) {
                 try {
                     c.close();
-                } catch (IOException e) {
-                    // When javac uses JDK 7 as a baseline, this code would be
-                    // better written to set any/all exceptions from all the
-                    // Closeables as suppressed exceptions on the FatalError
-                    // that is thrown.
+                } catch (Exception e) {
                     JCDiagnostic msg = diagFactory.fragment(Fragments.FatalErrCantClose);
                     throw new FatalError(msg, e);
                 }
```

Comment:
The comment is meant to suggest the following:

````
FatalError fe = null;
for (Closable c : closeables) {
    try {
        c.close();
    } catch (IOException e) {
        if (fe == null) {
            JCDiagnostic msg = diagFactory.fragment(Fragments.FatalErrCantClose);
            fe = new FatalError(msg):
        }
        fe.addSuppressed(e);
    }
}
if (fe != null) {
    throw fe;
}
````

(Apologies if there are any errors in that: I just typed it in directly.)


This is probably a case where you could probably get away with no test and use the label `norge-hard`.  If you were to write a test, it would have to involve creating multiple file system artifacts that cannot be closed, perhaps by creating wrappers that always throw exceptions when `close()` is called.

Reply:
Oops, I misunderstood the meaning of  `any/all exceptions from all the Closeables`.